File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int maxDepth = 0;
    public int maxDepth(TreeNode root) {
        
        if (root == null){
            return 0;
        }
        return dfs(root,1);
    }

    public int dfs(TreeNode node, int length){
        if (node == null){
            return length-1;
        }

        return  Math.max(dfs(node.left,length+1),dfs(node.right,length+1));

    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        
        
        maxPath = 0

        

        def dfs(root):
            nonlocal maxPath
            if not root:
                return 0
            
            left = dfs(root.left)
            right = dfs(root.right)

            totalCurrentLength = left+right
            maxPath = max(maxPath,totalCurrentLength)

            return max(left+1,right+1)
        dfs(root)
        return maxPath